pkg/datapath/linux/linux_defaults/linux_defaults.go

```diff
@@ -35,4 +35,7 @@ const (
 
 	// TunnelDeviceName the default name of the tunnel device when using vxlan
 	TunnelDeviceName = "cilium_vxlan"
+
+	// IPSec SPI value for node rules
+	IPSecNodeSPI = 2
```

Comment:
Is there anything magic about this SPI, particular reason `2` is chosen?

Reply:
Nothing magic about 2 it needs to avoid conflicting with (src,dst,spi) assigned to an endpoint range. Given two xfrm states S_1: (src_1, dst_1, spi_1)  and S_2 (src_2, dst_2, spi_2) the kernel considers the states S_1, S_2 to be colliding if dst_1 == dst_2 and spi_1 == spi_2. But we need to have a states to match outgoing traffic from both endpoints srcs and node ip srcs to another nodes dst ip. So we get these states S_1: (endpoint_ip_range, remote_ip, spi) and S_2: (node_ip, remote_ip, spi) so it collides with naive implementation with single spi*=1. Simple solution is to just use a special node spi here hence above magic number.

Reply:
And to make this more explicit I'll add a IPSecEndpointSPI = 1 constant and avoid the if clause and passing '0' as a magic default case through. I think it makes it a bit nicer.